File: Abstract Factory/PHP/TeamAssessmentConnectionFactory.php
<?php

require_once "IConnectionFactory.php";
require_once "TeamAssessmentRPCConnection.php";
require_once "TeamAssessmentHTTPConnection.php";

class TeamAssessmentConnectionFactory implements IConnectionFactory {
    public function createRPCConnection() {
        return new TeamAssessmentRPCConnection;
    }

    public function createHTTPConnection() {
        return new TeamAssessmentHTTPConnection;
    }
}

File: Abstract Factory/Python 3/TeamAssessmentConnectionFactory.py
from IConnectionFactory import IConnectionFactory
from TeamAssessmentRPCConnection import TeamAssessmentRPCConnection
from TeamAssessmentHTTPConnection import TeamAssessmentHTTPConnection

class TeamAssessmentConnectionFactory:
  def createRPCConnection(self):
    return TeamAssessmentRPCConnection()
  
  def createHTTPConnection(self):
    return TeamAssessmentHTTPConnection()